### ISSUE TITLE
Optional catch-all API route not working

### ISSUE THREAD

**Issue by milanku** [aiji42/next-qs-props#44]
Hi @aiji42

is there some way to support [Optional catch all API routes](https://nextjs.org/docs/api-routes/dynamic-api-routes#optional-catch-all-api-routes) ( [[...path]] )? I need to have routes

```
example.com/archive
example.com/archive?page=2&...
```

If I create project with structure like this

```
pages
|-- archive
  |-- _middleware.js
  |-- [...path].js
```

then `example.com/archive?page=2` works but `example.com/archive` returns 404. I have tried it like this as well with no success (`example.com/archive?page=2` triggers index.js with undefined query): 

```
pages
|-- archive
  |-- _middleware.js
  |-- [...path].js
  |-- index.js
```

Basically is there a way to add support for this? (I tried, but query is always undefined):

```
pages
|-- archive
  |-- _middleware.js
  |-- [[...path]].js
```

I guess for now I will redirect `example.com/archive` to `example.com/archive?page=1` and use the first file structure but it would be great if `[[...path]].js` worked.

**Comment by aiji42** [comment]
@milanku  Thanks for the question. This library does not support `Optional catch all API routes` at this time.
I am currently planning a major update (#34). Please wait for a while as the new version will include the support.

**Comment by aiji42** [comment]
v2.0.0 was released to support `Optional catch all API routes`.
Note, however, that prefetch navigation (e.g. Link component, router.push) is not supported in that case. We've also mentioned that in the readme.
This is due to the specs of Next.js, and we decided that it would be difficult to support it further at this time.
